include/h2o/httpclient.h

```diff
@@ -173,6 +173,30 @@ typedef struct st_h2o_httpclient_timings_t {
     struct timeval response_end_at;
 } h2o_httpclient_timings_t;
 
+/**
+ * a UDP tunnel established e.g., by successful CONNECT-UDP
+ */
+typedef struct st_h2o_httpclient_udp_tunnel_t {
+    /**
+     * closes the tunnel and discards the object
+     */
+    void (*destroy)(struct st_h2o_httpclient_udp_tunnel_t *tunnel);
+    /**
+     * The write callback. Contrary to what happens in the tcp tunnel, completion is immediate.
+     */
+    void (*write_)(struct st_h2o_httpclient_udp_tunnel_t *tunnel, const void *bytes, size_t len);
+    /**
+     * the on-read callback to be set by the user
+     */
+    void (*on_read)(struct st_h2o_httpclient_udp_tunnel_t *tunnel, const char *err, const void *bytes, size_t len);
```

Comment:
Same as above.

Reply:
Thanks for the suggestion. For reads, are you suggesting we use `recvmmsg` or is your idea that we change that signature to use iovecs?

Reply:
It's more about the tunnel API, to make sure that we have room for optimization (to send / receive packets in batch). Such optimizations could be sendmmsg / recvmmsg, UDP GSO / GRO, io_uring, or whatever.

As not all platforms have sendmmsg / recvmmsg, I do not think it's necessary for us to use the -**m**msg variant at this moment. Though doing a loop for at most 10 packets or something would not be a bad idea.